### PR TITLE
Closure externs and annotations

### PR DESCRIPTION
**Pull request by juj**
`--closure-externs` and `--closure-annotations` enable users to pass their own externs and annotations files to Closure build.

### CODE REVIEW

**Comment by hujiajie** [comment]
The [documentation](https://emscripten.org/docs/tools_reference/emcc.html#environment-variables) mentions that there's an environment variable `EMCC_CLOSURE_ARGS`. Will you be interested in a similar option, say `--closure-args=`, which would probably supersede the proposed ones?

**Comment by juj** [comment]
Ok, changed to `--closure-args`.

File: emcc.py [hunk 1]
```diff
@@ -2440,6 +2444,14 @@ def check_bad_eq(arg):
       options.llvm_lto = int(newargs[i + 1])
       newargs[i] = ''
       newargs[i + 1] = ''
+    elif newargs[i].startswith('--closure-args'):
+      check_bad_eq(newargs[i])
+      args = newargs[i + 1]
+      if (args[0] == '"' and args[-1] == '"') or (args[0] == "'" and args[-1] == "'"):
+        args = args[1:-1]
```
Comment (hujiajie) [inline emcc.py]: I doubt it's necessary to care about the quotation marks. If I type `"foo bar"` in the terminal, the quotation marks should be stripped by the shell and the string in sys.argv should be `foo bar`, isn't it?

Comment (juj) [reply emcc.py]: It's not necessary, but idea was to be more robust with command line and response file inputs. Probably excessive, dropped.

**Comment by kripken** [comment]
@juj looks like this broke post-commit CI on windows,

https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket.appspot.com/8906508831862804704/+/steps/Emscripten_testsuite__upstream__other_/0/stdout

It says
```
ERROR - Cannot read file C:bswirkinstallemscriptenteststest_closure_externs.js: C:bswirkinstallemscriptenteststest_closure_externs.js
```
Looks like the path separator is missing there?

**Comment by juj** [comment]
Thanks for pinging - posted https://github.com/emscripten-core/emscripten/pull/9123

**Pull request merged**